cumulus/parachains/integration-tests/emulated/tests/assets/asset-hub-westend/src/tests/reserve_transfer.rs

```diff
@@ -1132,8 +1145,445 @@ fn reserve_transfer_native_asset_from_para_to_para_through_relay() {
+	let native_asset = westend_system_emulated_network::asset_hub_westend_emulated_chain::asset_hub_westend_runtime::xcm_config::WestendLocationV3::get();
+	let asset_id = 9999u32;
+	let pool_asset = xcm::v3::Location::new(
+		0,
+		[
+			xcm::v3::Junction::PalletInstance(ASSETS_PALLET_ID),
+			xcm::v3::Junction::GeneralIndex(asset_id.into()),
+		],
+	);
+	let penpal_location = AssetHubWestend::sibling_location_of(PenpalA::para_id());
+	let penpal_sov_account = AssetHubWestend::sovereign_account_id_of(penpal_location.clone());
+
+	// Create SA-of-Penpal-on-AHR with ED.
+	// This ED isn't reflected in any derivative in a PenpalA account.
+	AssetHubWestend::fund_accounts(vec![(penpal_sov_account.clone().into(), ASSET_HUB_WESTEND_ED)]);
+
+	// Setup the pool between `native_asset` and `pool_asset` on AssetHub.
+	AssetHubWestend::execute_with(|| {
+		type RuntimeEvent = <AssetHubWestend as Chain>::RuntimeEvent;
+
+		// set up pool with asset_id <> native pair
+		assert_ok!(<AssetHubWestend as AssetHubWestendPallet>::Assets::create(
+			<AssetHubWestend as Chain>::RuntimeOrigin::signed(AssetHubWestendSender::get()),
+			asset_id.into(),
+			AssetHubWestendSender::get().into(),
+			ASSET_MIN_BALANCE,
+		));
+		assert!(<AssetHubWestend as AssetHubWestendPallet>::Assets::asset_exists(asset_id));
+
+		assert_ok!(<AssetHubWestend as AssetHubWestendPallet>::Assets::mint(
+			<AssetHubWestend as Chain>::RuntimeOrigin::signed(AssetHubWestendSender::get()),
+			asset_id.into(),
+			AssetHubWestendSender::get().into(),
+			10_000_000_000_000, // For it to have more than enough.
+		));
+
+		assert_ok!(<AssetHubWestend as AssetHubWestendPallet>::AssetConversion::create_pool(
+			<AssetHubWestend as Chain>::RuntimeOrigin::signed(AssetHubWestendSender::get()),
+			Box::new(native_asset),
+			Box::new(pool_asset),
+		));
+
+		assert_expected_events!(
+			AssetHubWestend,
+			vec![
+				RuntimeEvent::AssetConversion(pallet_asset_conversion::Event::PoolCreated { .. }) => {},
+			]
+		);
+
+		assert_ok!(<AssetHubWestend as AssetHubWestendPallet>::AssetConversion::add_liquidity(
+			<AssetHubWestend as Chain>::RuntimeOrigin::signed(AssetHubWestendSender::get()),
+			Box::new(native_asset),
+			Box::new(pool_asset),
+			1_000_000_000_000,
+			2_000_000_000_000, // `pool_asset` is worth half of `native_asset`
+			0,
+			0,
+			AssetHubWestendSender::get().into()
+		));
+
+		assert_expected_events!(
+			AssetHubWestend,
+			vec![
+				RuntimeEvent::AssetConversion(pallet_asset_conversion::Event::LiquidityAdded {lp_token_minted, .. }) => { lp_token_minted: *lp_token_minted == 1414213562273, },
+			]
+		);
+	});
+
+	let relay_asset_penpal_pov = Location::parent();
+	let custom_asset_penpal_pov = Location::new(
+		1,
+		[Parachain(1000), PalletInstance(ASSETS_PALLET_ID), GeneralIndex(asset_id.into())],
+	);
+	PenpalA::force_create_foreign_asset(
+		custom_asset_penpal_pov.clone(),
+		PenpalAssetOwner::get(),
+		false, // Asset not sufficient, has a pool to pay for fees.
```

Comment:
please stop associating "sufficiency" with fees paying ability - this is a source of confusion in the ecosystem which we too keep perpetuating.

`Sufficient` asset means it is allowed to cover account ED. Not sufficient means ED needs to be covered by some other asset.

If the asset it allowed to be used in fee paying is orthogonal to its sufficiency and depends solely on the chain's weight trader configuration.

```suggestion
		false,
```